### ISSUE TITLE
Could not connect to powchain endpoint

### ISSUE THREAD

**Issue by monossido** [diglos/pi-gen#13]
I boot the first time my raspberry with an ssd attached, I can see that it install a few things, then reboot.
But then I see this fail log on syslog, I think it's fatal, so geth and prysm is not syncing, right?

`Nov 30 19:50:30 ethnode-287315211 beacon-chain[1656]: time="2020-11-30 19:50:30" level=error msg="Could not connect to powchain endpoint" error="could not dial eth1 nodes: Post "http://localhost:8545": dial tcp [::1]:8545: connect: connection refused" prefix=powchain
`

I've the right port opened on my router

**Comment by Splurth** [comment]
Try to see if you can telnet locally on port  8545 try 127.0.0.1 and localhost, is there a difference ?

**Comment by monossido** [comment]
`ethereum@ethnode-287315211:~$ telnet 127.0.0.1 8545

Trying 127.0.0.1...

telnet: Unable to connect to remote host: Connection refused`

**Comment by diglos** [comment]
Can you type:

netstat -punta | grep 8545

**Comment by monossido** [comment]
`ethereum@ethnode-287315211:~$ sudo netstat -punta | grep 8545

ethereum@ethnode-287315211:~$ `

**Comment by diglos** [comment]
Ummh, seems that Geth is not listening to that port. Are you running geth with the --http flag?

**Comment by monossido** [comment]
With ps aux I cannot see geth, that's the problem probably..
At the beggining I was checkin this stuff after two days of uptime, so I was thinking that geth had finish its job.

But now it's a fresh installation and geth is not running.

Can you tell me how to launch it with the right argument and in background?
Thank you.

**Comment by monossido** [comment]
I think it's fixed by:


```
sudo systemctl start geth 
sudo systemctl enable geth
```


I leave the issu opened because geth was disabled on fresh install, don't know why.

**Comment by monossido** [comment]
I've also added "--http" to startup ARGS for geth, is that right? Why there is not that argoument by default?

Now prysm says:

```
Dec  1 18:33:24 ethnode-287315211 beacon-chain[1656]: time="2020-12-01 18:33:24" level=error msg="Could not connect to powchain endpoint" error="could not dial eth1 nodes: eth1 node using incorrect chain id, 0 != 1" prefix=powchain
```

**Comment by diglos** [comment]
It is really odd that geth wasn't enabled by default. Which image did you install?

**Comment by monossido** [comment]
This one
https://ethraspbian.com/downloads/ubuntu-20.04-preinstalled-server-arm64+raspi-eth2.img.zip

Also prysm did not have -pyrmont flag.

After this (and other) customization the raspberry is up and running in eth2 testnet (at the end I've used infuora service for the eth1 node actually)

**Comment by jahnludvik** [comment]
I have exactly the same problem, same outputs but I am no able to resolve it as @monossido

**Comment by monossido** [comment]
@jahnludvik I solved using infura service for eth1 node and configured prysm to use it.
Now my raspberry node is on mainnet working like a charm (Effectiveness 100%)

Though it's possible to run geth on the same raspberry I chose to not do that in order to save resources for prysm validator.

If you want to run geth and prysm on you own I suggest to understand how it works and configure it manually.
The img is not ready out of the box.

**Comment by raffobaffo** [comment]
@monossido  just made my day. --http flag was missing also in my image.

**Comment by diglos** [comment]
Hi, sorry for the lack of response here. Note that we didn't release an eth2 image for mainnet yet. We will next week along with a new documentation portal based on Read the Docs so hope all the config issues will be solved quickly.

@monossido, thanks for stopping by and help.

Thanks for your patience.

**Comment by jahnludvik** [comment]
Yeah thank you all guys. I guess I am waiting some days before retryin', then!

**Comment by Etherice** [comment]
Hello, I am trying to install and the first time after boot and I am getting the below error.

Oct 16 14:57:21 ethnode-a5cad6f85 beacon-chain[1686]: time="2021-10-16 14:57:21" level=error msg="Could not connect to powchain endpoint" error="could not dial eth1 nodes: Post "http://localhost:8545": dial tcp [::1]:8545: connect: connection refused" prefix=powchain

I am running into the same problem with geth but I am lost at how to add the http flag. Could somebody please help me out?
